### PR TITLE
added synthesiser choice in the makefile and a submodules/LIB/hardware/syn change

### PR DESCRIPTION
**Pull request by Vasco-Luz**
added a choice of synthesiser choise in the makefile. and Made a change in submodules/LIB/hardware/syn so it imports a build.mk in function of the passes synthesiser in the makefile

### CODE REVIEW

File: submodules/LIB/hardware/syn/genus/build.mk [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	@echo "set VSRC [glob $(VSRC)]" >> $@
+
+clean:
+	find . -maxdepth 1 -type f -not \( -name Makefile -o -name \*.tcl -o -name \*.sdc -o -name syn_build.mk \) -delete
+	@rm -rf ../../*.vh config.tcl fv results st_part_log
+ifneq ($(SYN_SERVER),)
+	ssh $(SYN_SSH_FLAGS) $(SYN_USER)@$(SYN_SERVER) 'if [ -f $(REMOTE_SYN_DIR)/Makefile ]; then make -C $(REMOTE_SYN_DIR) $@; fi'
+endif
+
+.PHONY: run clean config.tcl synth
```
Comment (jjts) [inline submodules/LIB/hardware/syn/genus/build.mk:61]: Confug.tcl is a file! Not a phony target.

File: submodules/LIB/hardware/syn/genus/build.mk [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ifneq ($(wildcard syn_build.mk),)
+include syn_build.mk
+endif
+
+OUTPUT_DIR=results
+
+#default node
+NODE ?= umc130
+
+SYN_SERVER=$(CADENCE_SERVER)
```
Comment (jjts) [inline submodules/LIB/hardware/syn/genus/build.mk:20]: This part should not be repeated for each tool. See simulation.

Comment (jjts) [reply submodules/LIB/hardware/syn/genus/build.mk:20]: See the FPGA folder, this is wrong . This way you need to repeat these commands for each syn tool.

File: submodules/LIB/hardware/syn/genus/build.mk (hunk 1 above)
Comment (Vasco-Luz) [reply submodules/LIB/hardware/syn/genus/build.mk:61]: I will change but that was in the original makefile.

File: submodules/LIB/hardware/syn/genus/build.mk (hunk 2 above)
Comment (Vasco-Luz) [reply submodules/LIB/hardware/syn/genus/build.mk:20]: OK will change, I done it like this because other sythesiser's maybe won't need a cadence server.

Comment (jjts) [reply submodules/LIB/hardware/syn/genus/build.mk:20]: This PR shows it was deleted from the makefile but it should not have been deleted

Why pass that code to the build.mk since it is not specific of the synt tool?


Close this PR and do a new one pls.

File: submodules/LIB/hardware/syn/genus/build.mk [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+SYN_SYNC_FLAGS=$(CADENCE_SYNC_FLAGS)
+
+synth: $(VHDR) $(VSRC)
+	genus -batch -files genus/build.tcl
+
+
+UFLAGS+=COV=$(COV)
+UFLAGS+=COV_TEST=$(COV_TEST)
+
+build: $(VHDR) $(VSRC) config.tcl
```
Comment (jjts) [inline submodules/LIB/hardware/syn/genus/build.mk:33]: all this code is unrelated to build.mk for genus
